pyre/cpython_tests/run.py

```
@@ -303,6 +308,35 @@ def death_signal(rc: int) -> str:
+
+    Every `Ran N tests` and the `skipped=N` of the verdict closing it are added
+    up rather than the last summary winning: a test file may run more than one
+    suite, and one that prints a captured child interpreter's output carries
+    that child's summary too. Totalling them keeps the caller's "nothing ran"
+    test conservative, since one suite that did run real tests outweighs
+    another that skipped everything.
+    """
+    ran = skipped = 0
+    counted = False
+    for line in f"{out}\n{err}".splitlines():
+        line = line.strip()
```

Ruff (0.16.1)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)